server/app/views/admin/BaseView.java

```diff
@@ -0,0 +1,241 @@
+package views.admin;
+
+import static autovalue.shaded.com.google.common.base.Preconditions.checkNotNull;
+
+import auth.ProfileUtils;
+import com.google.common.collect.ImmutableList;
+import controllers.AssetsFinder;
+import modules.ThymeleafModule;
+import org.thymeleaf.TemplateEngine;
+import play.mvc.Http;
+import services.settings.SettingsManifest;
+import views.CspUtil;
+import views.admin.shared.LayoutParams;
+import views.admin.shared.TemplateGlobals;
+import views.html.helper.CSRF;
+
+/**
+ * Base view class containing the structure and bare essentials to render the supplied Thymeleaf
+ * template.
+ *
+ * @param <TModel> A class or record that implements {@link BaseViewModel}
+ */
+public abstract class BaseView<TModel extends BaseViewModel> {
+  private static final String FAVICON_DATAURI =
```

Comment:
There's an existing issue for adding favicon to northstar. It can be added here too when ready, so for now just default to a hard-coded option.